### PR TITLE
sublime-text plugin support

### PR DESCRIPTION
**Pull request by ankitshubham97**
Addresses #21

Limitation: Only supported for macOS, verified on macOS Mojave version 10.14.5<period>

### CODE REVIEW

File: Default (OSX).sublime-keymap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[
```
Comment (lovesegfault) [inline Default (OSX).sublime-keymap]: Correct me if I'm wrong, but is the process of adding support for Linux just adding one of these keymap files for Linux, like this one is for OSX?

File: README.md [hunk 2]
```diff
@@ -148,6 +148,19 @@ command \
 ```
 This takes inspiration from the Eclipse feature.
 
+## How to install the sublime-text plugin
```
Comment (lovesegfault) [inline README.md]: Can we just list this on that Sublime package manager thing? I forget the name, I used it a long time ago.

File: README.md [hunk 3]
```diff
@@ -148,6 +148,19 @@ command \
...
+
+### OSX platform
```
Comment (lovesegfault) [inline README.md]: OSX -> MacOS ?

File: README.md [hunk 4]
```diff
@@ -148,6 +148,19 @@ command \
...
+1. Open Sublime.
+2. Go to "Preferences -> Browse Packages".
+3. This would open a Finder with the package path. For me, it opens up this path: "/Users/ankitshubham/Library/Application Support/Sublime Text 2/Packages/".
```
Comment (lovesegfault) [inline README.md]: What about Sublime 3?

File: beautysh_sublime_plugin.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		
+		## Uncomment follwing lines for printing stdout and stderr, might be
+		## helpful in debugging.
+		# print(stdout)
+		# print(stderr)
+
+		if p.returncode != 0:
+			print(p.returncode)
+			print(stderr)
+			print(stdout)
```
Comment (lovesegfault) [inline beautysh_sublime_plugin.py]: Newline at the end please :)

File: beautysh_sublime_plugin.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import sublime, sublime_plugin
+import subprocess
+
+class BeautyshCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		p = subprocess.Popen(                                                  \
```
Comment (lovesegfault) [inline beautysh_sublime_plugin.py]: I dislike how this is formatted, set the command array (`['python', ...]`) on a separate line, call it `beautysh_cmd` maybe, that should help make it smaller. Than just let autopep8 take care of it :)

File: README.md (hunk 2 above)
Comment (ankitshubham97) [reply README.md]: @lovesegfault I am aware of that (https://github.com/wbond/package_control_channel/). But for this, the code of the plugin must be present in this repo beforehand. Then, I just need to provide this information (url of this repo, name of the plugin etc) to the package control channel. I will take on that in the next phase after this commit is in.

File: Default (OSX).sublime-keymap (hunk 1 above)
Comment (ankitshubham97) [reply Default (OSX).sublime-keymap]: You are correct. But currently, I don't have access to a Linux system. So, I cannot verify if it works. However, I am certain that apart from creating another keymap file, there is nothing else that needs to be done in order to get this plugin working on a Linux system.

File: README.md (hunk 4 above)
Comment (ankitshubham97) [reply README.md]: It should work on Sublime-text 3 as well. This was just an example for the sake of clarity. I have Sublime-text 2 installed.

File: beautysh_sublime_plugin.py (hunk 6 above)
Comment (ankitshubham97) [reply beautysh_sublime_plugin.py]: Addressed!

**Comment by ankitshubham97** [comment]
@lovesegfault Addressed your review comments in 58fdce0

File: Default (OSX).sublime-keymap (hunk 1 above)
Comment (lovesegfault) [reply Default (OSX).sublime-keymap]: Alright, can you make a Linux file then, and I'll test it on my machine? I have a Sublime Text 3 license :)

File: README.md (hunk 2 above)
Comment (lovesegfault) [reply README.md]: Awesome, in this case I prefer we leave these instructions out until we can just tell the user to use PackageControl :)

File: README.md (hunk 4 above)
Comment (lovesegfault) [reply README.md]: Got it :)

File: beautysh_sublime_plugin.py (hunk 6 above)
Comment (lovesegfault) [reply beautysh_sublime_plugin.py]: Thanks!

File: README.md (hunk 2 above)
Comment (ankitshubham97) [reply README.md]: Sure!

File: Default (OSX).sublime-keymap (hunk 1 above)
Comment (ankitshubham97) [reply Default (OSX).sublime-keymap]: I added sublime-keymap for Linux in c161cd6.

Comment (ankitshubham97) [reply Default (OSX).sublime-keymap]: You can have a look at https://github.com/ankitshubham97/google-java-format-sublime/blob/master/README.md#linuxubuntu-platform to know how to manually install the plugin on Linux.

**Comment by ankitshubham97** [comment]
@lovesegfault I just tested it on Ubuntu 16.04 and it works fine.

**Comment by lovesegfault** [comment]
@ankitshubham97 Awesome, I just tested that it works for me as well :) 

Remove the instructions from the README while we don't have PackageControl support and I'll merge it ASAP :)

**Comment by ankitshubham97** [comment]
@lovesegfault Earlier, you commented:

> Awesome, in this case I prefer we leave these instructions out until we can just tell the user to use PackageControl :)
Maybe, I can add the instruction for Linux as well. Also, we can let the users know that currently, we do not have PackageControl support; so as of now, they need to follow those manual steps.

**Comment by lovesegfault** [comment]
@ankitshubham97 Yeah, I mentioned I preferred to _not_ provide instructions _until_ we can tell the user to use PackageControl.

I'll merge this and fix in master, and then worry about PackageControl :)

**Comment by ankitshubham97** [comment]
@lovesegfault Oh! I misread that thinking that you wanted to leave those instructions there in the README(not leaving out). I will soon raise a PR in PackageControl repo and record it in this thread.

**Comment by lovesegfault** [comment]
@ankitshubham97 Thank you!

**Pull request merged**